hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/ContractServiceImpl.java

```diff
@@ -40,18 +43,21 @@ public class ContractServiceImpl implements ContractService {
     private final ContractServiceComponent component;
 
     public ContractServiceImpl(@NonNull final AppContext appContext) {
-        this(appContext, null);
+        this(appContext, null, null);
     }
 
     public ContractServiceImpl(
-            @NonNull final AppContext appContext, @Nullable final Supplier<List<OperationTracer>> addOnTracers) {
+            @NonNull final AppContext appContext,
+            @Nullable final VerificationStrategies verificationStrategies,
+            @Nullable final Supplier<List<OperationTracer>> addOnTracers) {
         requireNonNull(appContext);
         this.component = DaggerContractServiceComponent.factory()
                 .create(
                         appContext.instantSource(),
                         // (FUTURE) Inject the signature verifier instance into the IsAuthorizedSystemContract
                         // C.f. https://github.com/hashgraph/hedera-services/issues/14248
                         appContext.signatureVerifier(),
+                        Optional.ofNullable(verificationStrategies).orElseGet(DefaultVerificationStrategies::new),
```

Comment:
(This is a clever way to get a concise syntax, but is it really worth it?)

Reply:
Haha I don't know, I just got used to this pattern for some reason. 🤷